File: DHT11.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
class DHT11 {
        private static String line;
	private static String[] data;
	static int humidity = 0;
	static int temperature = 0;
	public static void main(String[] args) throws Exception {
		 Runtime runTime = Runtime.getRuntime();
		 Process process = runTime.exec("python /home/pi/proiect/licenta/licenta/sensor.py");
		 BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(process.getInputStream()));
		 if((line = bufferedReader.readLine()) != null){

				 data = line.split(" ");
				 temperature = Integer.parseInt(data[0]);
				 System.out.println(data[0]);
				 humidity = Integer.parseInt(data[1]);
				 System.out.println(data[1]);
			  
		 } else { 
				 System.out.println("Data Error");
			}
	  
	 bufferedReader.close();
      	 process.waitFor();
      	 System.out.println("Temperature is : "+temperature+" 'C Humidity is :"+ humidity+" %RH");
      	 System.out.println("Done.");
	 }
}

File: dht11.py
import Adafruit_DHT
import time

sensor = Adafruit_DHT.DHT11

pin = 23
humidity, temperature = Adafruit_DHT.read_retry(sensor, pin)

while True:
	humidity, temperature = Adafruit_DHT.read_retry(sensor, pin)
	if humidity is not None and temperature is not None:	
		print temperature, humidity
	else:
		print('Failed to get reading. Try again!')
	time.sleep(.500)
